src/detect-pcre.c

```diff
@@ -70,15 +71,49 @@
 static int pcre_match_limit = 0;
 static int pcre_match_limit_recursion = 0;
 
-static pcre *parse_regex;
-static pcre_extra *parse_regex_study;
-static pcre *parse_capture_regex;
-static pcre_extra *parse_capture_regex_study;
+static DetectParseRegex parse_regex;
+static DetectParseRegex parse_capture_regex;
 
 #ifdef PCRE_HAVE_JIT
 static int pcre_use_jit = 1;
 #endif
 
+#ifdef PCRE_HAVE_JIT_EXEC
+#define PCRE_JIT_MIN_STACK 32*1024
+#define PCRE_JIT_MAX_STACK 512*1024
+
+static int g_jitstack_thread_ctx_id = -1;
+
+#endif
+
+/* \brief Helper function for using pcre_exec with/without JIT
+ */
+static int DetectPcreExec(DetectEngineThreadCtx *det_ctx, DetectParseRegex *regex, const char *str, const size_t strlen,
+                            int start_offset, int options,  int *ovector, int ovector_size)
+{
+    int ret;
+#ifdef PCRE_HAVE_JIT_EXEC
+    if (det_ctx && g_jitstack_thread_ctx_id != -1) {
+        pcre_jit_stack *jit_stack = (pcre_jit_stack *)
+            DetectThreadCtxGetKeywordThreadCtx(det_ctx, g_jitstack_thread_ctx_id);
+        if (jit_stack) {
+            ret = pcre_jit_exec(regex->regex, regex->study, str, strlen,
+                                start_offset, options, ovector, ovector_size,
+                                jit_stack);
+
+            if (ret == PCRE_ERROR_JIT_STACKLIMIT)
+                SCLogNotice("PCRE JIT stack limit exhausted");
```

Comment:
This seems risky, as an attacker might be able to craft input and cause (expensive) output.

Reply:
This is a left over debug message. Will remove.